components/ws-manager/pkg/manager/status.go

```diff
@@ -607,15 +609,14 @@ func (m *Manager) extractFailure(wso workspaceObjects, metrics *metrics) (string
 					phase = &c
 				}
 
-				// TODO(jenting)
-				// remove the debug log after we address the issue https://github.com/gitpod-io/gitpod/issues/12021
-				log.WithField("phase", status.Phase).WithField("code", terminationState.ExitCode).WithField("reason", terminationState.Reason).Infof("reason: %s, message: %s", status.Reason, status.Message)
-				// dump the underlying node taint
-				var node corev1.Node
-				if m.Clientset != nil {
-					if err := m.Clientset.Get(context.TODO(), types.NamespacedName{Namespace: "", Name: wso.NodeName()}, &node); err == nil {
-						for _, taint := range node.Spec.Taints {
-							log.WithField("nodeName", wso.NodeName()).Infof("taint key: %v, value: %v, effect: %v, time: %v", taint.Key, taint.Value, taint.Effect, taint.TimeAdded.String())
+				if terminationState.ExitCode == containerKilledExitCode && terminationState.Reason == "ContainerStatusUnknown" {
```

Comment:
Reaching here, has a workspace already completed backup?

Reply:
From the log we observed over the past seven days, the backup is completed `jsonPayload.conditions.finalBackupComplete=true`.

Reply:
Before ignoring this case, should we check `finalBackupComplete`?